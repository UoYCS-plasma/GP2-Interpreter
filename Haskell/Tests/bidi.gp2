Main = bidi

bidi () [
	(n1, empty)
	(n2, empty)
	(n3, empty)
|
	(e1 (B), n1, n2, empty)
	(e2 (B), n2, n3, empty)
	// should compile to a regular LUE
	(e3 (B), n1, n1, empty)
] => [
	(n1, empty # green )
	(n2, empty # green )
	(n3, empty # green )
|
	(e1 (B), n1, n2, empty)
	(e2 (B), n2, n3, empty)
	(e3 (B), n1, n1, empty)
]
interface = {n1, n2, n3}

