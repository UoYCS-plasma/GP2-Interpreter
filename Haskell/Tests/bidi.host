[
	(n1, empty)
	(n2, empty)
	(n3, empty)
|
	(e1, n1, n2, empty)
	(e2, n3, n2, empty)
	(e3, n1, n1, empty)
]
