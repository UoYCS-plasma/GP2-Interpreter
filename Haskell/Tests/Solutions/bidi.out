[
 (n1, empty # green)
 (n2, empty # green)
 (n3, empty # green)
|
 (e2, n1, n2, empty)
 (e3, n3, n2, empty)
 (e1, n3, n3, empty)
]
